modules/server/src/network/resolvers/aggregations.ts

```diff
@@ -1,101 +1,120 @@
+import { DocumentNode, GraphQLResolveInfo } from 'graphql';
+import graphqlFields from 'graphql-fields';
+import { config } from 'process';
 import { fetchGql } from '../gql';
-import { remoteConnectionQuery } from '../queries';
-import { NetworkAggregationConfig, SupportedNetworkFieldType } from '../types';
+import { supportedAggregationQueries } from '../queries';
+import { NetworkAggregationConfig } from '../types';
+import { ASTtoString } from '../util';
 
 /**
- * Returns an array of connection URLs
- * @param field
- * @param configs
- * @returns
+ * Returns only top level fields from a GQL request
+ *
+ * @param info GQL request info object
+ * @returns List of top level fields
  */
-const getConnectionURLs = (
-	field: SupportedNetworkFieldType,
-	configs: NetworkAggregationConfig[],
-): string[] => {
-	return configs
-		.filter((config) => config.supportedAggregations?.find((agg) => agg.name === field.name))
-		.map((config) => config.graphqlUrl);
+const getRootFields = (info: GraphQLResolveInfo) => {
+	const requestedFields = graphqlFields(info);
+	const fieldsAsList = Object.keys(requestedFields);
+	return fieldsAsList;
 };
 
 /**
- * Returns gql query based on type of field
- * @param fieldType
+ * Query remote connections and handle network responses
+ *
+ * @param query
  * @returns
  */
-const getGQLQuery = (fieldType: SupportedAggregation) => {
-	const query = remoteConnectionQuery.get(fieldType);
-	if (query) {
-		return query;
-	} else {
-		throw Error('Query not found for field');
-	}
-};
-
-/**
- * Queries remote connection gql server with query for field
- * @param url
- * @param fieldType
- * @returns
- */
-const queryRemoteConnection = async (url: string, fieldType: SupportedAggregation) => {
+const queryConnection = async (query: NetworkQuery) => {
+	const { url, gqlQuery } = query;
 	try {
-		const gqlQuery = getGQLQuery(fieldType);
-		const response = await fetchGql({ url, gqlQuery });
+		const response = await fetchGql({
+			url,
+			gqlQuery: ASTtoString(gqlQuery),
+		});
+
+		// axios response "data" field, graphql response "data" field
+		const responseData = response.data?.data;
 		if (response.status === 200 && response.statusText === 'OK') {
-			const responseData = response.data?.data;
 			return responseData;
 		}
-		throw Error(`Query to ${url} failed`);
+
+		console.error('unexpected response data');
 	} catch (error) {
-		console.error(error);
+		/**
+		 * TODO: expand on error handling for instance of Axios error for example
+		 */
+		console.error(`network failure`, error);
 		return;
 	}
 };
 
-/**
- * TODO: Placeholder for resolving multiple remote connection responses into single aggregation
- */
-const resolveAggregation = (response: any, type: string) => {};
+export const queryConnections = async (queries: NetworkQuery[]) => {
+	const networkQueryPromises = queries.map(async (query) => {
+		const response = await queryConnection(query);
+		return { config, response };
```

Comment:
Where is `config` coming from? Why is it being added to each object in this array?